modules/gapi/src/backends/streaming/gstreamingbackend.hpp

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright 2020 Intel Corporation.
+//
+// This software and the related documents are Intel copyrighted materials,
+// and your use of them is governed by the express license under which they
+// were provided to you ("License"). Unless the License provides otherwise,
+// you may not use, modify, copy, publish, distribute, disclose or transmit
+// this software or the related documents without Intel's prior written
+// permission.
+//
+// This software and the related documents are provided as is, with no
+// express or implied warranties, other than those that are expressly
+// stated in the License.
+//
+
+#ifndef OPENCV_GAPI_COPY_KERNEL_HPP
+#define OPENCV_GAPI_COPY_KERNEL_HPP
+
+#include <opencv2/gapi/gkernel.hpp>
+#include <opencv2/gapi/streaming.hpp>
+#include "gstreamingkernel.hpp"
+
+namespace cv {
+namespace gimpl {
+
+class RMatMediaBGRAdapter : public cv::RMat::Adapter
+{
+    cv::MediaFrame m_frame;
+public:
+    RMatMediaBGRAdapter(cv::MediaFrame frame) : m_frame(frame) { };
+
+    virtual cv::RMat::View access(cv::RMat::Access a) override
+    {
+        auto view = m_frame.access(a == cv::RMat::Access::W ? cv::MediaFrame::Access::W
+                                                            : cv::MediaFrame::Access::R);
+
+        return cv::RMat::View(desc(),
+                              reinterpret_cast<uchar*>(view.ptr[0]),
+                              view.stride[0],
+                              [=](){});
+    }
```

Comment:
So view gets destroyed here immediately.

Can you guarantee it will be captured by `lambda` and not optimized out?

Lambda body is empty here.

Reply:
I'd destroy a view manually in the callback, maybe by `swap(view, {})` or by releasing an unique ptr.

Reply:
Any actions here?

Reply:
> There are some missing changes to allow MediaFrame be the output object (compared to this #18987).



> There are some missing changes to allow MediaFrame be the output object (compared to this #18987).

What kind of changes do you mean ?

Reply:
Resolved via `shared_ptr`

Reply:
@TolyaTalamanov some of our backends utilize `OCVCall` or something like that from cpu backend. Added a couple of things there to support MediaFrame as output

Reply:
The callback now is a `mutable` lambda though